### PR TITLE
Readme tweaks

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Extract Api.md so it would be more easy to include it with hapijs.com plugin pages.

### CODE REVIEW

**Comment by cjihrig** [comment]
Does this need a `<!-- tocstop -->` like the joi document uses?

**Comment by AdrieanKhisbe** [comment]
Not necessarily if we maintain the toc by hand.

Tell me then if I should remove it or no.

**Comment by cjihrig** [comment]
I have no opinion. I hate the API.md idea. So, whatever makes it work properly is fine with me.

**Comment by Marsup** [comment]
Toc generation is independent from the API.md decision, though it has helped me a lot consistently getting the toc right and avoiding useless PRs. Your choice.

**Comment by AdrieanKhisbe** [comment]
I added then the toc generation from joi.

File: package.json [hunk 1]
```diff
@@ -16,11 +16,14 @@
         "hoek": "3.x.x"
     },
     "devDependencies": {
-        "lab": "7.x.x"
+        "lab": "8.x.x",
+        "markdown-toc": "^0.12.3"
```
Comment (Marsup) [inline package.json]: Can't guarantee it'll keep working with a 0.x version, this should be made sticky to 0.12 as joi did.

Comment (AdrieanKhisbe) [inline package.json]: you mean keep lab 7?

Comment (cjihrig) [inline package.json]: No, markdown-toc. Semver doesn't really apply the same way prior to v1.x.x. Can you drop the ^

Comment (AdrieanKhisbe) [inline package.json]: okey dokey.
Didn't know about that. 

Shall I made a dedicated PR?

Comment (cjihrig) [inline package.json]: Yes please.

Comment (AdrieanKhisbe) [inline package.json]: voilà: https://github.com/hapijs/code/pull/55

**Pull request merged**